src/pages/RFOX/components/Stake/StakeInput.tsx

```diff
@@ -25,116 +50,451 @@ const formControlProps = {
+  const handleRuneAddressChange = useCallback(
+    (address: string | undefined) => {
+      onRuneAddressChange(address)
+    },
+    [onRuneAddressChange],
+  )
 
   const handleWarning = useCallback(() => {
     setShowWarning(true)
   }, [])
 
   const handleSubmit = useCallback(() => {
+    if (!(stakingAssetAccountId && runeAddress && isValidStakingAmount)) return
+
+    setConfirmedQuote({
+      stakingAssetAccountId,
+      stakingAssetId,
+      stakingAmountCryptoBaseUnit: toBaseUnit(amountCryptoPrecision, stakingAsset?.precision ?? 0),
+
+      runeAddress,
+    })
     history.push(StakeRoutePaths.Confirm)
-  }, [history])
+  }, [
+    stakingAsset?.precision,
+    amountCryptoPrecision,
+    history,
+    isValidStakingAmount,
+    runeAddress,
+    setConfirmedQuote,
+    stakingAssetAccountId,
+    stakingAssetId,
+  ])
 
   const assetSelectComponent = useMemo(() => {
-    return <TradeAssetSelect assetId={asset?.assetId} isReadOnly onlyConnectedChains={true} />
-  }, [asset?.assetId])
+    return (
+      <TradeAssetSelect assetId={stakingAsset?.assetId} isReadOnly onlyConnectedChains={true} />
+    )
+  }, [stakingAsset?.assetId])
+
+  const stakingAssetBalanceFilter = useMemo(
+    () => ({
+      accountId: stakingAssetAccountId ?? '',
+      assetId: stakingAssetId,
+    }),
+    [stakingAssetAccountId, stakingAssetId],
+  )
+  const stakingAssetBalanceCryptoPrecision = useAppSelector(state =>
+    selectPortfolioCryptoPrecisionBalanceByFilter(state, stakingAssetBalanceFilter),
+  )
+
+  const feeAssetBalanceFilter = useMemo(
+    () => ({
+      accountId: stakingAssetAccountId ?? '',
+      assetId: feeAsset?.assetId,
+    }),
+    [feeAsset?.assetId, stakingAssetAccountId],
+  )
+  const feeAssetBalanceCryptoPrecision = useAppSelector(state =>
+    selectPortfolioCryptoPrecisionBalanceByFilter(state, feeAssetBalanceFilter),
+  )
+  const stakingAssetFiatBalance = bnOrZero(stakingAssetBalanceCryptoPrecision)
+    .times(stakingAssetMarketData.price)
+    .toString()
+
+  // Simple memoized value vs. react-hook-form error because of the chicken and egg situation of this being dependant of fees being loaded,
+  // and having to be reset when going from not enough fee balance to not enough balance.
+  const hasEnoughFeeBalanceError = useMemo(() => {
+    const fees = approvalFees || stakeFees
+
+    const hasEnoughFeeBalance = bnOrZero(fees?.networkFeeCryptoBaseUnit).lte(
+      toBaseUnit(feeAssetBalanceCryptoPrecision, feeAsset?.precision ?? 0),
+    )
 
-  if (!asset) return null
+    if (!hasEnoughFeeBalance)
+      return translate('modals.send.errors.notEnoughNativeToken', { asset: feeAsset?.symbol })
+
+    return
+  }, [
+    approvalFees,
+    feeAsset?.precision,
+    feeAsset?.symbol,
+    feeAssetBalanceCryptoPrecision,
+    stakeFees,
+    translate,
+  ])
+
+  const amountFieldInputRules = useMemo(() => {
+    return {
+      defaultValue: '',
+      validate: {
+        hasEnoughBalance: (input: string) => {
```

Comment:
We can do one better and have the `hasEnoughBalanceForFees` here as well - yes, this is chicken-and-eggy and we cannot consume the outer scope fees query here, but we can fire it with `queryClient` here with the same queryKey/cache/gc options here, meaning outer scope will leverage cache on the next render cycle after input values are updated

Reply:
Actually meh, that would be a lot better conceptually, but not worth it with the boilerplate

Reply:
*akschually*, wasn't that bad to leverage component reactivity using trigger - c23bc603575f660a33571234806047542b1c46a8 and obligatory retest jam https://jam.dev/c/aecd3ceb-8f77-430f-a8c5-e4b4b87ca633